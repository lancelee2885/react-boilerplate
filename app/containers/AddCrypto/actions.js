/*
 *
 * AddCrypto actions
 *
 */

import {
  CHANGE_FORM_DATA,
  SUBMIT_FORM_DATA,
  SUBMIT_FORM_DATA_ERROR,
  SUBMIT_FORM_DATA_SUCCESS,
} from './constants';

/**
 * Changes the input fields of the form
 */
export function changeFormData(formData) {
  return {
    type: CHANGE_FORM_DATA,
    formData,
  };
}

/**
 * Changes the input fields of the form
 */
export function submitFormData() {
  return {
    type: SUBMIT_FORM_DATA,
  };
}

/**
 * Changes the input fields of the form
 */
export function formDataSubmitted(formData) {
  return {
    type: SUBMIT_FORM_DATA_SUCCESS,
    formData,
  };
}

/**
 * Changes the input fields of the form
 */
export function formDataError(err) {
  return {
    type: SUBMIT_FORM_DATA_ERROR,
    err,
  };
}
