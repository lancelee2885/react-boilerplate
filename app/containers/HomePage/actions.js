/*
 * App Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */

import {
  LOAD_CRYPTO,
  LOAD_CRYPTO_ERROR,
  LOAD_CRYPTO_SUCCESS,
} from './constants';

/**
 * Load the crypto from backend, this action starts the request saga
 *
 * @return {object} An action object with a type of LOAD_CRYPTO
 */
export function loadCryptos() {
  return {
    type: LOAD_CRYPTO,
  };
}

/**
 * Dispatched when the cryptocurrencies are loaded by the request saga
 *
 * @param  {array} cryptos The crypto data
 *
 * @return {object}      An action object with a type of LOAD_CRYPTO_SUCCESS passing the cryptos
 */
export function cryptosLoaded(cryptos) {
  return {
    type: LOAD_CRYPTO_SUCCESS,
    cryptos,
  };
}

/**
 * Dispatched when loading the cryptocurrencies fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of LOAD_CRYPTO_ERROR passing the error
 */
export function cryptossLoadingError(error) {
  return {
    type: LOAD_CRYPTO_ERROR,
    error,
  };
}
