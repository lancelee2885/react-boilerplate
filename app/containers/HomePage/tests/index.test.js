/**
 * Test the HomePage
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import { HomePage, mapDispatchToProps } from '../index';
import configureStore from '../../../configureStore';
import { loadCryptos } from '../actions';

describe('<HomePage />', () => {
  const cryptos = [
    {
      symbol: 'testSymbol',
      name: 'testName',
      description: 'testDescription',
      iconURL: 'testURL',
    },
  ];
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <HomePage
            loading={false}
            error={false}
            cryptos={[]}
            loadCryptosProp={() => null}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should try to fetch crypto on first mount', () => {
    const fetchSpy = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <HomePage
            loading={false}
            error={false}
            cryptos={cryptos}
            loadCryptosProp={fetchSpy}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(fetchSpy).toHaveBeenCalled();
  });

  describe('mapDispatchToProps', () => {
    describe('loadCryptosProp', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.loadCryptosProp).toBeDefined();
      });

      it('should dispatch loadCryptos when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.loadCryptosProp();
        expect(dispatch).toHaveBeenCalledWith(loadCryptos());
      });
    });
  });
});
