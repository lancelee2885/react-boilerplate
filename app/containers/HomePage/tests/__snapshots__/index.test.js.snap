// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<div
  class="Wrapper__ItemsContanier-vsnnp1-0 bquUeX"
>
  <div
    class="Wrapper__ItemsWrapper-vsnnp1-1 knjxCi"
  />
</div>
`;
